lib/app.js

```diff
@@ -202,3 +207,33 @@ function getProperty(target, prop) {
   }
   return member;
 }
+
+
+function bindMessenger(Application, agent) {
+  const agentMessenger = agent.messenger;
+  return class MeesengerApplication extends Application {
+    constructor(options) {
+      super(options);
+
+      const appMessenger = this.messenger;
+      agentMessenger.send = new Proxy(agentMessenger.send, {
+        apply(target, thisArg, [ action, data ]) {
+          setImmediate(() => appMessenger._onMessage({ action, data }));
+        },
+      });
+    }
+    get messenger() {
+      return this[MESSENGER];
+    }
+    set messenger(m) {
+      m.send = new Proxy(m.send, {
+        apply(target, thisArg, [ action, data ]) {
+          setImmediate(() => agentMessenger._onMessage({ action, data }));
+        },
+      });
+      this[MESSENGER] = m;
+    }
+
+    get [Symbol.for('egg#eggPath')]() { return path.dirname(__dirname); }
```

Comment:
Yes.

The reason that eggPath should be defined is every frameworks extended egg-core should tell it where the framework is. But in this case, it's no use.